lib/internal/url.js

```diff
@@ -804,6 +804,16 @@ class URL {
     this.#updateContext(bindingUrl.parse(input, base));
   }
 
+  static parse(input, base = undefined) {
```

Comment:
This is an extremely slow implementation. Because we are catching an error, but in reality we don't need to catch it. If you use a private `Symbol` and pass it to URL constructor and depending on the existence of the Symbol, not throw the error and `privately` (without exposing to the user) create a way to understand if the parsing is correct or not, it would be a lot faster.

Reply:
Should we have benchmarks?

This new function should be no slower.

Reply:
> This new function should be no slower.

One of the [original reasons for `canParse`](https://github.com/whatwg/url/issues/713#issuecomment-1282329130) is that exceptions are rather expensive.

The `parse` function is basicaly just `canParse` but returns the url/null instead of just a boolean, so it definitely makes sense to use the performant option here.

Reply:
For an initial implementation as long as it's correct and doesn't lead to e.g. backport woes, I think it's fine to consider performance in a follow up, especially when it comes from a first-time contributor. As far as "extremely slow" goes, it doesn't seem to be any worse than implementing `structuredClone()` on top of `MessagePort`s (what the initial implementation did), yet that also looked fine to me as an initial implementation, as that also came from a new contributor.

Reply:
@joyeecheung @anonrig thanks for the suggestion,

I have tried another approach as you mentioned to use symbol and pass it to the constructor, I updated the node_url source file `parse` function as it checks raise_exception boolean then throws error otherwise not.

is this better or overdoing?

Reply:
> is this better or overdoing?

I don't have an opinion from a node maintainers point of view as I'm not one, though if you want a simpler pure JS version just using `canParse` is still about 20 times faster than `try/catch` in a quick profile I did.

i.e.:

```js
function parse(url, base) {
    if (URL.canParse(url, base)) {
        return new URL(url, base);
    }
    return null;
}
```